client/signup/controller.js

```diff
@@ -112,6 +113,15 @@ export default {
 					const countryCode = geo.data.body.country_short;
 					if ( 'gutenberg' === abtest( 'newSiteGutenbergOnboarding', countryCode ) ) {
 						window.location.replace( window.location.origin + '/new' + window.location.search );
+					} else if ( 'variantPasswordless' === abtest( 'passwordlessAfterPlans', countryCode ) ) {
```

Comment:
The `/start/personal`, `/start/premium`, `/start/business`, and `/start/ecommerce` flows will also redirect to the passwordless flow. We don't want that because these flows do not have a plans step whereas the passwordless flow does. 

There are two ways to fix this:
1. In flows.js [in this line](https://github.com/Automattic/wp-calypso/blob/master/client/signup/config/flows.js#L120), we could return the passwordless flow name if user is in the variant. This means we can delete all the changes made to client/signup/controller.js. 
2. Another way is to add flow name checks here. So:
```
if ( 'gutenberg' === abtest( 'newSiteGutenbergOnboarding', countryCode ) ) {
    // do something
} else if ( 
    context.pathname.indexOf( 'personal' ) === -1 &&
    context.pathname.indexOf( 'premium' ) === -1 &&
    context.pathname.indexOf( 'business' ) === -1 &&
    context.pathname.indexOf( 'ecommerce' ) === -1 &&
    'variantPasswordless' === abtest( 'passwordlessAfterPlans', countryCode )
) {
    // do something.
}
```

Reply:
I think we should go with the second approach as the countryCode provided in that context has proven to be more reliable.